### PR TITLE
Gutenboarding - header adjustments

### PR DESCRIPTION
**Pull request by chriskmnds**
#### Changes proposed in this Pull Request

Some adjustments to header content. See https://github.com/Automattic/wp-calypso/issues/40138

#### Testing instructions

http://calypso.localhost:3000/gutenboarding?fresh

Fixes https://github.com/Automattic/wp-calypso/issues/40138

#### TODO

- [x] "Start your website" content.
- [x] Align elements horizontally.
- [x] Extend domain selector with "... is available".
- [ ] Anything about "premium" in `[premiumdomain] is available` that needs to be addressed? Should this somehow only apply to premium stuff?
- [x] `W` icon dimensions
- [ ] Move "Create my site" button (https://github.com/Automattic/wp-calypso/issues/40138#issuecomment-599508709)

### CODE REVIEW

File: client/landing/gutenboarding/components/header/style.scss [hunk 1]
```diff
@@ -36,10 +36,10 @@
 
 .gutenboarding__header-group {
 	display: flex;
-	flex-direction: column;
+	flex-direction: row;
 	margin: 10px;
 	font-size: 14px;
-	align-items: flex-start;
+	align-items: center;
```
Comment (sirreal) [inline client/landing/gutenboarding/components/header/style.scss]: I _think_ we can roll back these changes.

File: client/landing/gutenboarding/components/header/index.tsx [hunk 2]
```diff
@@ -215,7 +218,8 @@ const Header: FunctionComponent = () => {
 					<Icon icon="wordpress-alt" className="gutenboarding__header-wp-icon" />
 				</div>
 				<div className="gutenboarding__header-group">
-					{ siteTitle ? (
+					{ siteTitleElement }
+					{ siteTitle && (
```
Comment (sirreal) [inline client/landing/gutenboarding/components/header/index.tsx]: Does this work? I think this is how the structure would naturally do this.

```suggestion
					{ siteTitleElement }
				</div>
				<div className="gutenboarding__header-group">
					{ siteTitle && (
```

Comment (chriskmnds) [reply client/landing/gutenboarding/components/header/index.tsx]: > Does this work?

As far as I can see, it looks like (considering the accompanied CSS changes).

> I think this is how the structure would naturally do this.

If I understand correctly, you referring to `gutenboarding__header-group` being a `column-based group` as the expected structure for header groups? In that case, then yes, we can put those into individual header-groups and it should be more natural by that standard. fwiw these were grouped together, so it seemed to make sense to keep them grouped together - but no objection otherwise.

Comment (chriskmnds) [reply client/landing/gutenboarding/components/header/index.tsx]: Is it a good question why a header group should be column-based by default, and whether that can/should be parameterised instead?

Comment (chriskmnds) [reply client/landing/gutenboarding/components/header/index.tsx]: Also, as far as I can see there is only one instance of a "header-group" with more than one element (the above). Is this correct? If that's the case, then I guess we can induce that a "header-group" should instead be "row based" by default, not the other way. Unless there is some other consensus I am not familiar with.

Comment (sirreal) [reply client/landing/gutenboarding/components/header/index.tsx]: The designs and layout have changed considerably since this was built. It's progressed towards a less cluttered header.

If the layout no longer serves our purposes then I'm happy to see it changed so that it does.

I haven't looked closely at this, it jumped out as a surprising change. It seems to work within the layout on one hand (same HTML structure) but change how the layout is laid out. I suspect there's a lot of possible simplification if the existing structure isn't well adapted.

In summary, I expected either only HTML structure changes or a larger overhaul of the layout. This seemed like a very small change on both ends which likely means the layout isn't a good fit and that it will no longer behave as was intended.

I'm trying to provide helpful thoughts, they're not meant to be blocking.

Comment (chriskmnds) [reply client/landing/gutenboarding/components/header/index.tsx]: Yep, understood. I will take another look at this and maybe do a better job at refactoring. I don't think either/or (my changes, your suggestion) make a lot of sense if these two elements (title, domain-select) are not naturally grouped. If they are, then splitting these apart pretty much makes the concept of "group" obsolete - they all become single-item groups, and potentially the only related things remain ungrouped. At the same time, applying a style to reverse in the only case that is used sounds disconnected. 

Anyway, looking at the designs - there seems to be an upcoming "group" candidate (the lang selector), which is also "row based" by default.

**Comment by chriskmnds** [comment]
Let's get this in before conflicts start to prop and we'll reiterate on #40321
